src/marqo/s2_inference/clip_utils.py

```diff
@@ -182,15 +193,31 @@ def download_image_from_url(image_path: str, media_download_headers: dict, timeo
     headers.update(media_download_headers)
     c.setopt(pycurl.HTTPHEADER, [f"{k}: {v}" for k, v in headers.items()])
 
+    # callback to check file size for video and audio
+    if modality in [Modality.VIDEO, Modality.AUDIO]:
+        max_size = read_env_vars_and_defaults_ints(EnvVars.MARQO_MAX_VIDEO_AUDIO_SEARCH_FILE_SIZE)
+        def progress(download_total, downloaded, upload_total, uploaded):
+            if downloaded > max_size:
+                return 1
```

Comment:
What error did we get when this happens? Is user going to get a user-friendly error message indicating the the media size exceeds the threshold?

Reply:
Yes, the error message will be `error_message = f"Media file `{image_path}` exceeds the maximum allowed size for {modality}."`